src/inmanta/server/services/resourceservice.py

```diff
@@ -328,6 +328,25 @@ async def get_resource_increment_for_agent(self, env: data.Environment, agent: s
 
         now = datetime.datetime.now().astimezone()
 
+        def on_agent(res: ResourceIdStr) -> bool:
+            idr = Id.parse_id(res)
+            return idr.get_agent_name() == agent
+
+        # This is a bit subtle
+        # Any resource we consider deployed has to be marked as such
+        # Otherwise the agent will fail the deployment
```

Comment:
We could fix this by considering available as deployed, but I do like this second layer of defense in the agent